hypertrace-gradle-docker-plugin/src/main/java/org/hypertrace/gradle/docker/DockerPlugin.java

```diff
@@ -86,15 +86,29 @@ private void addDockerBuildTaskForImage(
   }
 
   private void addDefaultTags(Project project, DockerPluginExtension extension) {
-    extension.tag(this.getVersionString(project));
-    extension.tag(DockerTag.LATEST, tag -> tag.onlyIf(unused -> extension.tagLatest.get()));
+    if (this.isReleaseVersion(project)) {
+      extension.tag(this.getVersionString(project));
+      extension.tag(DockerTag.LATEST, tag -> tag.onlyIf(unused -> extension.tagLatest.get()));
+    }
+    extension.tag(this.getBranchTag());
   }
 
   private String getVersionString(Project project) {
     return project.getVersion()
                   .toString();
   }
 
+  private boolean isReleaseVersion(Project project) {
+    return !this.getVersionString(project)
+                .contains("SNAPSHOT");
+  }
+
+  private String getBranchTag() {
+    // Use the value of CIRCLE_BRANCH environment variable if defined (that is, the branch name used
```

Comment:
sorry to nit, but this rephrases the optional, which is literally correct, but doesn't add value I meant. if not saying "for example, "master"" It is likely better to not have a comment at all.

My reason for asking is as I don't know our CI setup, I don't know if we routinely build branches besides master. If the net effect of this change is that we end up with "master" tag, either here or in the CHANGELOG instead is probably more helpful from a product POV even if I agree what you are saying is technically correct.

Reply:
I actually had the example in here and removed it because of the technicality. First, the docker tag is actually used both in pull requests and a master build today, so even pull requests would _technically_ use this (I think it's fine not to explain that case). The part I was hesitating more on is that I don't believe the default branch is consistent across all repositories. I know we've been starting to move repository default branches from `master` to `main`. As that happens, we'll end up in a state where the tags are different across different containers - not ideal. So I tried to account for that here - maybe it would be better to just always use some known label rather than the branch name then?

Reply:
I had no idea about the move from master to main.. such a thing would need to be sorted (including if main is a good name, and here isn't a good place to discuss that).

In any case, I think there is a flex to things like this which is why some policy around tagging and which branches to build or tag end up in config, as filtering before executing a build can be more explicit and sometimes easier to manage.